### PR TITLE
TDR-2238 call addFileStatus mutation to set upload status as 'Success'

### PR DESCRIPTION
**Pull request by vimleshtna**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala [hunk 1]
```diff
@@ -6,7 +6,10 @@ import java.util.UUID
 
 import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification.S3EventNotificationRecord
 import com.typesafe.config.{Config, ConfigFactory}
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala:8]: Think this is an un-used import: `import com.typesafe.config.{Config, ConfigFactory}`

File: src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala [hunk 2]
```diff
@@ -43,6 +46,23 @@ class FileUtils()(implicit val executionContext: ExecutionContext, keycloakDeplo
...
+      token <- keycloakUtils.serviceAccountToken(lambdaConfig("auth.client.id"), lambdaConfig("auth.client.secret"))
+      response <- client.getResult(token, document, Option(afs.Variables(AddFileStatusInput(fileId, statusType, statusValue))))
+      fileStatus <- getFileStatusResponse(response, statusType, statusValue)
+    } yield fileStatus
+  }
+
+  private def getFileStatusResponse(response: GraphQlResponse[afs.Data], statusType: String, statusValue: String): Future[AddFileStatus] = response.errors match {
+    case Nil => Future(response.data.get.addFileStatus)
+    case List(authorisedError: NotAuthorisedError) => failed(authorisedError.message)
+    case errors => failed(s"Unable to add file status with statusType '$statusType' and value '$statusValue'. Errors: ${errors.map(e => e.message).mkString}")
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala:63]: Might be worth adding the file id into the error message to make tracking down any issues easier.

File: src/main/scala/uk/gov/nationalarchives/downloadfiles/Lambda.scala [hunk 3]
```diff
@@ -93,6 +96,8 @@ class Lambda {
             value("fileId", fileId),
             value("consignmentId", consignmentId)
           )
+          // add FileStatus with statusType as 'Upload' and value as 'Success'
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/downloadfiles/Lambda.scala:99]: I don't think this comment is needed.

File: src/test/scala/uk/gov/nationalarchives/downloadfiles/FileUtilsTest.scala [hunk 4]
```diff
@@ -179,6 +181,67 @@ class FileUtilsTest extends AnyFlatSpec with MockitoSugar with ScalaFutures  {
     res.getMessage shouldEqual "GraphQL response contained errors: General error"
   }
 
+  "The addFileStatus method" should "add a file status" in {
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/downloadfiles/FileUtilsTest.scala:184]: Can remove the `The` from the test title.

File: src/test/scala/uk/gov/nationalarchives/downloadfiles/FileUtilsTest.scala [hunk 5]
```diff
@@ -179,6 +181,67 @@ class FileUtilsTest extends AnyFlatSpec with MockitoSugar with ScalaFutures  {
...
+    status.statusType should equal("Upload")
+    status.statusValue should equal("Success")
+    status.fileId should equal(fileID)
+
+    val expectedId = "client_id"
+    val expectedSecret = "secret"
+    verify(keycloakUtils).serviceAccountToken(expectedId, expectedSecret)
+  }
+
+  "The addFileStatus method" should "error if the graphql query returns not authorised errors" in {
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/downloadfiles/FileUtilsTest.scala:208]: Can remove the `The` from the test title.

File: src/test/scala/uk/gov/nationalarchives/downloadfiles/FileUtilsTest.scala [hunk 6]
```diff
@@ -179,6 +181,67 @@ class FileUtilsTest extends AnyFlatSpec with MockitoSugar with ScalaFutures  {
...
+        List(), List(), Some(Extensions(Some("NOT_AUTHORISED")))))))
+    when(client.getResult[Identity](any[BearerAccessToken], any[Document], any[Option[afs.Variables]])(any[SttpBackend[Identity, Any]], any[ClassTag[Identity[_]]]))
+      .thenReturn(Future.successful(graphqlResponse))
+
+    val res: Throwable = FileUtils().addFileStatus(keycloakUtils, client, uuid, lambdaConfig).failed.futureValue
+
+    res.getMessage shouldEqual "Not authorised message"
+  }
+
+  "The addFileStatus method" should "error if the graphql query returns a general error" in {
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/downloadfiles/FileUtilsTest.scala:228]: Can remove the `The` from the test title.

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments.

File: src/test/scala/uk/gov/nationalarchives/downloadfiles/FileUtilsTest.scala (hunk 4 above)
Comment (vimleshtna) [reply src/test/scala/uk/gov/nationalarchives/downloadfiles/FileUtilsTest.scala:184]: Yeah, we can remove it. Actually, all other test methods start with 'The', so I did in the same way.

File: src/main/scala/uk/gov/nationalarchives/downloadfiles/Lambda.scala (hunk 3 above)
Comment (vimleshtna) [reply src/main/scala/uk/gov/nationalarchives/downloadfiles/Lambda.scala:99]: Yeah, it's not required. I can remove it. Thanks.

File: src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala (hunk 1 above)
Comment (vimleshtna) [reply src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala:8]: Removed. Thanks.

File: src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala (hunk 2 above)
Comment (vimleshtna) [reply src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala:63]: Yeah, make sence, thanks

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**